orttraining/orttraining/training_api/optimizer.h

```diff
@@ -66,9 +102,25 @@ struct Optimizer {
 
   Status Step();
 
+  /**
+   * @brief Get the current optimizer state.
+   *
+   * Be noted the returned optimizer_checkpoint_states will hold new references to
+   * original momentum states.
+   * @return Status
+   */
   Status GetStateDict(OptimizerCheckpointState& optimizer_checkpoint_states);
 
-  Status LoadStateDict(const OptimizerCheckpointState& optimizer_checkpoint_states);
+  /**
+   * @brief Load states from optimizer_checkpoint_states into current optimizer state.
+   *
+   * Be noted Optimizer will reuse the data buffer of passed in optimizer_checkpoint_states.
+   * If the device of momentums in optimizer_checkpoint_states is not matching its parameter device,
+   * an implicit copy will be done during the LoadStateDict, but reserving using the original OrtValue
+   * with the copied data buffer.
+   * @return Status
+   */
+  Status LoadStateDict(OptimizerCheckpointState& optimizer_checkpoint_states);
```

Comment:
Does this need to be public?

Reply:
Originally there is a test need it. Just refactored the test, and moved LoadStateDict into private.